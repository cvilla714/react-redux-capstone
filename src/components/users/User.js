/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-undef */
/* eslint-disable camelcase */
/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Spinner from '../layout/Spinner';

class User extends Component {
  componentDidMount() {
    this.props.getUser(this.props.match.params.login);
  }

  render() {
    const {
      name,
      company,
      avatar_url,
      location,
      bio,
      blog,
      login,
      html_url,
      followers,
      following,
      public_repos,
      public_gists,
      hireable,
    } = this.props.user;

    const { loading } = this.props;

    if (loading) return <Spinner />;

    return (
      <>
        <Link to="/" className="btn btn-light">
          Back to Search
        </Link>
        Hireable:
        {hireable ? (
          <i className="fas fa-check text-success" />
        ) : (
          <i className="fas fa-times-circle text-danger" />
        )}
        <div className="card grid-2">
          <div className="all-center">
            <img
              src={avatar_url}
              className="round-img"
              alt=""
              style={{ width: '150px' }}
            />
            <h1>{name}</h1>
            <p>Location: {location}</p>
          </div>
          <div>
            {bio && (
              <>
                <h3>Bio</h3>
                <p>{bio}</p>
              </>
            )}
            <a href={html_url} className="btn btn-dark my-1">
              Visit Github Profiel
            </a>
            <ul>
              <li>
                {login && (
                  <>
                    <strong>Username: </strong>
                    {login}
                  </>
                )}
              </li>
              <li>
                {company && (
                  <>
                    <strong>Compnay: </strong>
                    {company}
                  </>
                )}
              </li>
              <li>
                {blog && (
                  <>
                    <strong>Webiste: </strong>
                    {blog}
                  </>
                )}
              </li>
            </ul>
          </div>
        </div>
        <div className="card text-center">
          <div className="badge badge-primary">Followers:{followers}</div>
          <div className="badge badge-success">Following:{following}</div>
          <div className="badge badge-light">Public Repos:{public_repos}</div>
          <div className="badge badge-dark">Public Gists:{public_gists}</div>
        </div>
      </>
    );
  }
}

export default User;
