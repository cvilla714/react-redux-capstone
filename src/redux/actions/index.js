import axios from 'axios';
import {
  CLEAR_USERS,
  GET_REPOS,
  GET_USER,
  SEARCH_USERS,
  SET_LOADING,
  LOGS_ERROR,
} from './types';

export const setLoading = () => ({
  type: SET_LOADING,
});

export const searchUsers = (text) => async (dispatch) => {
  try {
    setLoading();

    const res = await axios.get(
      `https://api.github.com/search/users?q=${text}`,
      {
        headers: {
          Authorization: `token ${process.env.REACT_APP_GITHUB_TOKEN}`,
        },
      },
    );
    dispatch({
      type: SEARCH_USERS,
      payload: res.data.items,
    });
  } catch (error) {
    dispatch({
      type: LOGS_ERROR,
      payload: error.respose.statusText,
    });
  }
};

export const getUser = (username) => async (dispatch) => {
  try {
    setLoading();

    const res = await axios.get(`https://api.github.com/users/${username}`, {
      headers: {
        Authorization: `token ${process.env.REACT_APP_GITHUB_TOKEN}`,
      },
    });

    dispatch({
      type: GET_USER,

      payload: res.data,
    });
  } catch (error) {
    dispatch({
      type: LOGS_ERROR,
      payload: error.respose.statusText,
    });
  }
};

export const getUserRepos = (username) => async (dispatch) => {
  try {
    setLoading();

    const res = await axios.get(
      `https://api.github.com/users/${username}/repos?per_page=5&sort=created:asc`,
      {
        headers: {
          Authorization: `token ${process.env.REACT_APP_GITHUB_TOKEN}`,
        },
      },
    );

    dispatch({
      type: GET_REPOS,
      payload: res.data,
    });
  } catch (error) {
    dispatch({
      type: LOGS_ERROR,
      payload: error.respose.statusText,
    });
  }
};

export const clearUsers = () => ({
  type: CLEAR_USERS,
});
