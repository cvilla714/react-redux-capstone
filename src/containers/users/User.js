import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Repos from '../repos/Repos';
import Spinner from '../../components/layout/Spinner';
import { getUser, getUserRepos } from '../../components/actions/index';

const User = ({
  user, loading, repos, getUser, getUserRepos, match,
}) => {
  useEffect(() => {
    getUser(match.params.login);
    getUserRepos(match.params.login);
  }, []);

  const {
    name,
    company,
    avatar_url: avatarUrl,
    location,
    bio,
    blog,
    login,
    html_url: htmlUrl,
    followers,
    following,
    public_repos: publicRepos,
    public_gists: publicGists,
    hireable,
  } = user;

  if (loading) return <Spinner />;

  return (
    <>
      <Link to="/" className="btn btn-light">
        Back to Search
      </Link>
      Hireable:
      {hireable ? (
        <i className="fas fa-check text-success" />
      ) : (
        <i className="fas fa-times-circle text-danger" />
      )}
      <div className="card grid-2">
        <div className="all-center">
          <img
            src={avatarUrl}
            className="round-img"
            alt=""
            style={{ width: '150px' }}
          />
          <h1>{name}</h1>
          <p>
            Location:
            {location}
          </p>
        </div>
        <div>
          {bio && (
            <>
              <h3>Bio</h3>
              <p>{bio}</p>
            </>
          )}
          <a href={htmlUrl} className="btn btn-dark my-1">
            Visit Github Profiel
          </a>
          <ul>
            <li>
              {login && (
                <>
                  <strong>Username: </strong>
                  {login}
                </>
              )}
            </li>
            <li>
              {company && (
                <>
                  <strong>Compnay: </strong>
                  {company}
                </>
              )}
            </li>
            <li>
              {blog && (
                <>
                  <strong>Webiste: </strong>
                  {blog}
                </>
              )}
            </li>
          </ul>
        </div>
      </div>
      <div className="card text-center">
        <div className="badge badge-primary">
          Followers:
          {followers}
        </div>
        <div className="badge badge-success">
          Following:
          {following}
        </div>
        <div className="badge badge-light">
          Public Repos:
          {publicRepos}
        </div>
        <div className="badge badge-dark">
          Public Gists:
          {publicGists}
        </div>
      </div>
      <Repos repos={repos} />
    </>
  );
};

User.defaultProps = {
  loading: false,
  match: '',
  repos: [],
  getUser: PropTypes.func,
  getUserRepos: PropTypes.func,
};

User.propTypes = {
  user: PropTypes.objectOf(Array).isRequired,
  loading: PropTypes.bool,
  match: PropTypes.string,
  repos: PropTypes.arrayOf(Array),
  getUser: PropTypes.func,
  getUserRepos: PropTypes.func,
};
const mapStateToProps = (state) => ({
  user: state.github.user,
});

export default connect(mapStateToProps, { getUser, getUserRepos })(User);
