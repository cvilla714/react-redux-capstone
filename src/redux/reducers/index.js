import { combineReducers } from 'redux';
import githubReducer from './githubReducer';
import alertReducer from './alertReducer';
import filterReducer from './filterReducer';

export default combineReducers({
  github: githubReducer,
  alert: alertReducer,
  filter: filterReducer,
});
