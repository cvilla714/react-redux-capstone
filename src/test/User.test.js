import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { MemoryRouter } from 'react-router-dom';
import combineReducers from '../reducers';
import User from '../components/users/User';

test('User component renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <MemoryRouter>
        <User />
      </MemoryRouter>
      ,
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
