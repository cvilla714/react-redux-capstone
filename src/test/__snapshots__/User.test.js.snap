// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders correctly 1`] = `
Array [
  <a
    className="btn btn-light"
    href="/"
    onClick={[Function]}
  >
    Back to Search
  </a>,
  "Hireable:",
  <i
    className="fas fa-times-circle text-danger"
  />,
  <div
    className="card grid-2"
  >
    <div
      className="all-center"
    >
      <img
        alt=""
        className="round-img"
        style={
          Object {
            "width": "150px",
          }
        }
      />
      <h1 />
      <p>
        Location:
      </p>
    </div>
    <div>
      <a
        className="btn btn-dark my-1"
      >
        Visit Github Profiel
      </a>
      <ul>
        <li />
        <li />
        <li />
      </ul>
    </div>
  </div>,
  <div
    className="card text-center"
  >
    <div
      className="badge badge-primary"
    >
      Followers:
    </div>
    <div
      className="badge badge-success"
    >
      Following:
    </div>
    <div
      className="badge badge-light"
    >
      Public Repos:
    </div>
    <div
      className="badge badge-dark"
    >
      Public Gists:
    </div>
  </div>,
  ",",
]
`;
